cranelift-faerie/src/backend.rs

```diff
@@ -87,6 +247,174 @@ impl FaerieCompiledFunction {
     }
 }
 
+struct CFIEncoder<'a> {
+    reg_map: &'a DwarfRegMapper<'a>,
+    cfa_def_reg: Option<isa::RegUnit>,
+    cfa_def_offset: Option<isize>,
+}
+
+struct DwarfRegMapper<'a> {
+    isa: &'a Box<dyn TargetIsa>,
+    reg_info: RegInfo,
+}
+
+impl<'a> DwarfRegMapper<'a> {
```

Comment:
I mentioned this on irc a few days ago but `DwarfRegMapper` could arguably live in cranelift-codegen as it's fundamentally an `int->int` function, and doesn't _need_ to return a `gimli::Register`. This and `CFIEncoder` might also make sense in `cranelift-module` as they are, pulling in a `gimli` dependency there and not in `cranelift-codegen`. @yurydelendik would `cranelift-module` be a useful place to share this logic translating `FrameLocationChange` to CFI instructions for your use in wasmtime?

Reply:
Issue #729 is about the register mapping.

Reply:
I've moved this to `cranelift-module` and noted some reasoning in that issue. I'll leave this unresolved until we decide here or there what to do with `DwarfRegMapper`.

Reply:
>  @yurydelendik would cranelift-module be a useful place to share this logic translating FrameLocationChange to CFI instructions for your use in wasmtime?

@iximeow yes, and in fact, I would like to have entire cratelift-module change for other reasons.